test/rules/use_key_in_widget_constructors.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N use_key_in_widget_constructors`
+
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+class NoConstructorWidget extends StatefulWidget { // LINT
+}
+
+class _PrivateWidget extends StatefulWidget { // OK
+}
+
+class MyWidget extends StatelessWidget {
+  MyWidget(); // LINT
+  MyWidget.withKey({Key key}); // OK
```

Comment:
Do we want to verify in this case that `key` is being passed to the superclass?

Reply:
we could.